public/main.js

```diff
@@ -74,9 +80,27 @@ function reloadTodoList() {
     }
     todoListPlaceholder.style.display = "block";
     getTodoList(function(todos) {
+        var filteredTodos;
         todoListPlaceholder.style.display = "none";
         updateIncompletes(todos, countLabel);
-        todos.forEach(function(todo) {
+        switch (filter) {
+            case FILTER_ALL:
```

Comment:
Haha I told him not to do that to distinguish behaviour for invalid filters. Up to you Jamie!